spec/views/post_index_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe 'posts#index', type: :feature do
+  before do
+    @user1 = User.create(name: 'Benjamin', photo: 'photo_url1', bio: 'I am a software developer.', postsCounter: 3)
+    @post1 = Post.create(author: @user1, title: 'Hello', text: 'Good.', commentssCounter: 3, likesCounter: 4)
+  end
+
+  scenario 'display the username of the user' do
+    visit user_posts_path(@user1)
+    expect(page).to have_content(@user1.name)
+  end
+
+  scenario 'display the profile picture of the user' do
+    visit user_posts_path(@user1)
+    expect(page).to have_css("img[src='photo_url1']")
+  end
+
+  scenario 'display the number of posts the user has written' do
+    visit user_posts_path(@user1)
+    expect(page).to have_content("Number of posts: #{@user1.postsCounter}")
+  end
+
+  scenario 'display the post title' do
+    visit user_posts_path(@user1)
+    expect(page).to have_content(@post1.title)
+  end
+
+  scenario 'display some of the post body' do
+    visit user_posts_path(@user1)
+    expect(page).to have_content(@post1.text)
+  end
+
+  scenario 'display the first comments on a post' do
```

Comment:
- [ ] Hey! 🌟 Quick heads-up – I noticed that all the test cases in the post index are hitting a snag. 👀 Take a peek at the errors in the screenshot below – they're causing a bit of a hiccup. 🚨 How about giving those a once-over and working your coding magic to fix them up? 🛠️ Your debugging skills got this! Happy coding! 🚀

![image](https://github.com/BilalLiaquat7/blogapp/assets/112261237/0101375d-cb3a-4747-8d78-8ecbb0017f38)